lib/entry/crossDCsStreamPersistence.js

```diff
@@ -0,0 +1,275 @@
+        }
+
+    }
+
+    removeEntry(entryId) {
+
+        logger.debug(`====> [removeEntry] removing [${entryId}] from persistence list`);
+        if (this._entriesToProcess.hasOwnProperty(entryId)) {
+            delete this._entriesToProcess[entryId];
+            delete this._playlistsData[entryId];
+
+            /*        // todo: rethink flash operation location and timing...
+             return this.flushEntryList(this._entriesToProcess)
+             .catch((err) => {
+             logger.error(`failed to get entry server nodes for stopped entry [${entryId}], error: [${ErrorUtils.error2string(err)}]`);
+             return Q.resolve(true);
+             });*/
+            if (Object.keys(this._entriesToProcess).length === 0) {
+                return this.flushEntryList(this._entriesToProcess)
+                    .catch((err) => {
+                        logger.info(`====> [removeEntry] error flushing stopped entries. Error: [${ErrorUtils.error2string(err)}]`);
+                    })
+                    .then(() => {
+                      return Q.resolve(true);
+                    });
+            }
+            return Q.resolve(true);
+        } else {
+            return Q.resolve(true);
+        }
+    }
+
+    // initialized persistence list (stopped entries).
+    init() {
+        logger.debug(`====> [init] restoring persistence list from ${stoppedEntriesPath}`);
+        return qio.exists(stoppedEntriesPath)
+            .then((exists) => {
+                if (exists) {
+                    return qio.read(this.stoppedEntriesFileName);
+                } else {
+                    return Q.reject(false);
+                }
+            })
+            .then((stoppedEntries) => {
+                let pAll = _.map(JSON.parse(stoppedEntries), (entryId) => {
+                    this.addEntry(entryId);
+                });
+                return Q.allSettled(pAll);
+            })
+            .catch((err) => {
+                if (_.isObject(err)) {
+                    if (err.code != 'ENOENT') {
+                        logger.info(`====> [init] error restoring stopped entries list, error: [${ErrorUtils.error2string(err)}]`);
+                        return Q.reject(err);
+                    } else {
+                        logger.warn(`${this.stoppedEntriesFileName} was not found`);
+                        return Q.resolve();
+                    }
+                } else {
+                    return qio.makeDirectory(stoppedEntriesPath)
+                        .catch((err) => {
+                            logger.error(`====> [init] error restoring stopped entries list, error: [${ErrorUtils.error2string(err)}]`);
+                            return Q.reject(err);
+                        });
+
+                }
+            });
+
+    }
+
+    flushEntryList() {
+
+        logger.debug(`====> [flushEntryList] updating persistence list on disk`);
+
+        if (Object.keys(this._entriesToProcess).length > 0) {
```

Comment:
why all this code?
we can serialize the _entriesToProcess even if they are empty, this will simplify and remove lots of code